### PR TITLE
Update tests to v2

### PR DESCRIPTION
**Pull request by jason-fox**

### CODE REVIEW

**Comment by fgalan** [comment]
With regards to files mentioned in removed code:

* test/examples/deviceProvisioning/expectedProvisioningRequest.json
* test/examples/deviceProvisioning/expectedDataUpdateRequest.json
* test/examples/deviceProvisioning/expectedProvisioningPluginRequest.json
* test/examples/deviceProvisioning/expectedDataUpdatePluginRequest.json 
* test/examples/ngsi-communication/expectedDeviceRegisterRequest.json
* test/examples/ngsi-communication/expectedDeviceUpdateDataRequest.js

I understand the correspond to NGSIv-1 payloads, so they should be also removed in this PR.

**Comment by fgalan** [comment]
With regards to the coverage drop in this PR (-0.7%), looking to https://coveralls.io/builds/41390644 maybe is due to?

imagen

So it seems that after changing the test the "happy case" (measure is received) has broken.

Does it make sense? Could you have a look, please?

**Comment by jason-fox** [comment]
> I understand the correspond to NGSIv-1 payloads, so they should be also removed in this PR.

Fixed ab96be2

**Comment by mapedraza** [comment]
> With regards to the coverage drop in this PR (-0.7%), looking to https://coveralls.io/builds/41390644 maybe is due to?
> 
> imagen
> 
> So it seems that after changing the test the "happy case" (measure is received) has broken.
> 
> Does it make sense? Could you have a look, please?

After checking the result on that part of the code performing the tests with https://github.com/jason-fox/sigfox-iotagent/tree/ngsi-v1 branch, It seems it is a problem related with tests translations to V2. A PR to Jason's fork fixing tests was created here: https://github.com/jason-fox/sigfox-iotagent/pull/1

After merging the PR the coverage should increase to the previous value.

**Comment by mapedraza** [comment]
Now, after reparing the tests, no one was failing, so it is needed to add a new test for the error handling scenario to recover al the coverage.

File: test/unit/ngsi-communication-test.js [hunk 1]
```diff
@@ -200,12 +206,12 @@ describe('Context Broker communication', function() {
...
+        it('should raise a controlled error', function (done) {
+            request(options, function (error, response, body) {
                 should.not.exist(error);
                 response.statusCode.should.equal(404);
                 done();
             });
         });
     });
-});
+});
```
Comment (mapedraza) [inline test/unit/ngsi-communication-test.js:217]:
```suggestion

        describe('When a sigfox measure fails updating Context Broker', function () {
            const options = {
                url: 'http://localhost:' + config.sigfox.port + '/update',
                method: 'GET',
                qs: {
                    id: 'sigApp1',
                    time: 1430909015,
                    statin: '0A5F',
                    lng: -4,
                    lat: 41,
                    data: '000000020000000000230c6f'
                }
            };
    
            it('should raise a controlled error', function (done) {
                request(options, function (error, response, body) {
                    should.not.exist(error);
                    response.statusCode.should.equal(200);
                    done();
                });
            });
        });
    });
});
```

Adding a failing test to cover the line was uncovered after fixing the tests as mentioned [here](https://github.com/telefonicaid/sigfox-iotagent/pull/111#issuecomment-919153077)

**Comment by jason-fox** [comment]
After merge and lint 016f7ff

> Coverage remained the same at 82.593%

🎉

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**